# HG changeset patch
# User Rik <rik@octave.org>
# Date 1407285991 25200
#      Tue Aug 05 17:46:31 2014 -0700
# Branch stable
# Node ID 082063dc2165c71bb25457c7cb7db81fdb068595
# Parent  1197f04eec09708e5890065ea6dff322d3306d66
ls.m: Fix bug listing directories with spaces in name (bug #42920).

* ls.m: Escape spaces in directory names before passing to system command.

diff --git a/scripts/miscellaneous/ls.m b/scripts/miscellaneous/ls.m
--- a/scripts/miscellaneous/ls.m
+++ b/scripts/miscellaneous/ls.m
@@ -59,17 +59,17 @@ function retval = ls (varargin)
   endif
 
   if (nargin > 0)
     args = tilde_expand (varargin);
     if (ispc () && ! isunix ())
       ## shell (cmd.exe) on MinGW uses '^' as escape character
       args = regexprep (args, '([^\w.*? -])', '^$1');
     else
-      args = regexprep (args, '([^\w.*? -])', '\\$1');
+      args = regexprep (args, '([^\w.*?-])', '\\$1');
     endif
     args = sprintf ("%s ", args{:});
   else
     args = "";
   endif
 
   cmd = sprintf ("%s %s", __ls_command__, args);
 
